cmd/argocd-application-controller/commands/argocd_application_controller.go

```diff
@@ -211,8 +217,20 @@ func getClusterFilter() func(cluster *v1alpha1.Cluster) bool {
 			shard, err = sharding.InferShard()
 			errors.CheckError(err)
 		}
+		distributionFunction := sharding.GetShardByIdUsingHashDistributionFunction()
 		log.Infof("Processing clusters from shard %d", shard)
-		clusterFilter = sharding.GetClusterFilter(replicas, shard)
+		db := db.NewDB(settingsMgr.GetNamespace(), settingsMgr, kubeClient)
+		log.Infof("Using filter function:  %s", shardingAlgorithm)
+		switch {
+		case shardingAlgorithm == common.RoundRobinShardingAlgorithm:
+			distributionFunction = sharding.GetShardByIndexModuloReplicasCountDistributionFunction(db, shardingAlgorithm)
+		case shardingAlgorithm == common.LegacyShardingAlgorithm:
+		default:
+			distributionFunctionName := runtime.FuncForPC(reflect.ValueOf(distributionFunction).Pointer())
+			log.Warnf("No distribution function named '%s' found. Defaulting to '%s'", shardingAlgorithm, distributionFunctionName)
+		}
```

Comment:
It seems that this switch is a duplicated logic with `sharding.GetDistributionFunction`. Is is possible to reuse that code here?

Reply:
Indeed, it seems to be a bad copy paste that happened during rebases and refactoring. I have fixed that.